packages/tools/viewer/src/viewer.ts

```diff
@@ -221,11 +213,61 @@ function reduceMeshesExtendsToBoundingInfo(maxExtents: Array<{ minimum: Vector3;
  * @param models The models to consider when computing the bounding info
  * @returns The computed bounding info for the models or null
  */
-function computeModelsBoundingInfos(models: Model[]): Nullable<ViewerBoundingInfo> {
+function computeModelsBoundingInfos(models: readonly Model[]): Nullable<ViewerBoundingInfo> {
     const maxExtents = computeModelsMaxExtents(models);
     return reduceMeshesExtendsToBoundingInfo(maxExtents);
 }
 
+/**
+ * Generates a HotSpot from a camera by computing its spherical coordinates (alpha, beta, radius) relative to a target point.
+ *
+ * The target point is determined using the camera's forward ray:
+ *   - If the ray intersects with a mesh in the model, the intersection point is used as the target.
+ *   - If no intersection is found, a fallback target is calculated by projecting the distance
+ *     between the camera and the model's center along the camera's forward direction.
+ *
+ * @param model The reference model used to determine the target point.
+ * @param camera The camera from which the HotSpot is generated.
+ * @returns A HotSpot object.
+ */
+export async function CreateHotSpotFromCamera(model: Model, camera: Camera): Promise<HotSpot> {
+    await import("core/Culling/ray");

```

Comment:
I assume this is the only reason for the function to be async?
I don't see any cleaner solution, apart from maybe taking this import to a higher level (for example making the user import it and provide the `Ray` class as function variable). Not sure how good this is TBH. just a suggestion.

Reply:
Yes that's right. I'd like to keep this super simple for users, so I'll leave it as is.